codemagic.yaml

```diff
@@ -53,19 +69,77 @@ workflows:
     <<: *configure_environment
     triggering:
       events:
-        # run when a pull request is opened or updated
+        # Run when a pull request is opened or updated
         - pull_request
       branch_patterns:
-        # review changes BEFORE they’re merged into any branches
+        # Review changes BEFORE they’re merged into any branches
         - pattern: '*'
           source: false
-        # review changes AFTER they have been merged into develop branch
+        # Will not run on develop branch as it is already covered by build-and-deploy-template-compose-staging
         - pattern: 'develop'
-        # review changes AFTER they have been merged into main branch
+          include: false
+        # Will not run on main branch as it is already covered by build-and-deploy-template-compose-production
         - pattern: 'main'
+          include: false
       cancel_previous_builds: true
     scripts:
       - *detekt_on_template_xml
       - *unit_test_on_template_xml
     artifacts:
       - *artifacts_template_xml
+
+  build-and-deploy-template-compose-staging:
+    name: Build and deploy template-compose staging to Firebase App Distribution
+    <<: *configure_environment
+    triggering:
+      events:
+        - push
+      branch_patterns:
+        - pattern: develop
+    scripts:
+      - *set_up_google_services_files_from_environment_variables
+      - *detekt_on_template_compose
+      - *unit_test_on_template_compose
+      - name: Build APK for staging
+        working_directory: ./template-compose
+        script: |
+          ./gradlew assembleStagingDebug -PversionCode=$BUILD_NUMBER
+    artifacts:
+      - *artifacts_template_compose
+      - *artifacts_staging_apk
+    publishing:
+      firebase:
+        firebase_service_account: $FIREBASE_SERVICE_ACCOUNT_CREDENTIALS
+        android:
+          app_id: 1:215362241811:android:51e902760fcff1243446e6
```

Comment:
to make the workflow to be more flexible, this could be fetched from an env, such as `FIREBASE_APP_ID_STAGING`